syndeo-template/src/main/java/com/google/cloud/syndeo/common/ProviderUtil.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.syndeo.common;
+
+import com.google.cloud.syndeo.v1.SyndeoV1.ConfiguredSchemaTransform;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.apache.beam.sdk.schemas.Schema.FieldType;
+import org.apache.beam.sdk.schemas.SchemaTranslation;
+import org.apache.beam.sdk.schemas.transforms.SchemaTransform;
+import org.apache.beam.sdk.schemas.transforms.SchemaTransformProvider;
+import org.apache.beam.sdk.values.PCollectionRowTuple;
+import org.apache.beam.sdk.values.Row;
+
+public class ProviderUtil {
+
+  public static final Map<String, SchemaTransformProvider> PROVIDERS = loadProviders();
+
+  /** Load providers, including bringing in one for SchemaIOs. */
+  private static Map<String, SchemaTransformProvider> loadProviders() {
+    ServiceLoader<SchemaTransformProvider> provider =
+        ServiceLoader.load(SchemaTransformProvider.class);
+    List<SchemaTransformProvider> list =
+        StreamSupport.stream(provider.spliterator(), false).collect(Collectors.toList());
+    list.addAll(SchemaIOTransformProviderWrapper.getAll());
+
+    Map<String, SchemaTransformProvider> map = new HashMap<>();
+    for (SchemaTransformProvider p : list) {
+      map.put(p.identifier(), p);
+    }
+
+    return map;
+  }
+
+  /** Return a list of all providers. */
+  public static Collection<SchemaTransformProvider> getProviders() {
+    return PROVIDERS.values();
+  }
+
+  /** Get a specific provider. */
+  public static SchemaTransformProvider getProvider(String name) {
+    return PROVIDERS.getOrDefault(name, null);
+  }
+
+  /** A class for storing an io / configuration pair. */
+  public static class TransformSpec {
+    public String inputId;
+    public Row configuration = null;
+    public SchemaTransformProvider provider = null;
+
+    public TransformSpec(String inputId, List<Object> configurationAsList) {
+      this.inputId = inputId;
+      provider = getProvider(inputId);
+      if (provider == null) throw new RuntimeException(inputId + " not found ");
```

Comment:
Normally we always add the braces, even when it's just a one-line statement, so:

```
if (provider == null) {
  throw...
}
```

Also, wouldn't `IllegalArgumentException` be better here, since it is more specific than `RuntimeException`?

Same goes for the other cases.

Reply:
agreed. thanks. done!

Reply:
Still seeing the original. Did you forget to push a commit?

Reply:
oops fixed thanks!

Reply:
(I had fixed it in a line below that's why!)